### ISSUE TITLE
Problem all_joint_states with all_joint_names

### ISSUE THREAD

**Issue by SinaiAranda** [Danfoa/invite-robotics#5]
Hi,
I'm work with Motoman dual robot sda20d, I take your package like reference for my [package](https://github.com/sinaiaranda-CIDESI/motoman_sda20d). I add this [configuration](https://github.com/sinaiaranda-CIDESI/motoman_sda20d/tree/master/motoman_sda20d_support/config), and I run [roslaunch](https://github.com/sinaiaranda-CIDESI/motoman_sda20d/blob/master/motoman_sda20d_moveit_config/launch/robot_interface_streaming_sda20d.launch) and this just run one time and show:

[FATAL] [1525701109.760904112]: ASSERTION FAILED
**	file = /home/sda20/cidesi_robots_ws/src/motoman-indigo-devel/motoman_driver/src/industrial_robot_client/joint_relay_handler.cpp**
**	line = 320**
**	cond = all_joint_state.positions.size() == all_joint_names.size()**

[joint_state-2] process has died [pid 7228, exit code -5, cmd /home/sda20/cidesi_robots_ws/devel/lib/motoman_driver/robot_state __name:=joint_state __log:=/home/sda20/.ros/log/c9baf556-51fd-11e8-8a16-009a9e9d9347/joint_state-2.log].
log file: /home/sda20/.ros/log/c9baf556-51fd-11e8-8a16-009a9e9d9347/joint_state-2.log*

In joint_states show correctly sda20d/sda20d_r1_controller/join_states, sda20d/sda20d_r2_controller/join_states, sda20d/sda20d_b1_controller/join_states except sda20d/sda20d_b2_controller/join_states (this don't show).

The display of the robot (trough of MotoPlus) show: 
Controller number of group = 3.
State Server Send failure. Closing state server connection.
Closing Motion Server Connection.
Motion Server Connection Closed.

Any help I would like to thank you.

**Comment by Danfoa** [comment]
Hi @sinaiaranda-CIDESI  sorry for the late answer I have been a little busy lately.
I am sorry to say that we didnt face this problem but we did face a similar issue that I reported on ros-industrial/motoman#209, check the answer from the mantainer, and the pull request he references, this seems to be a problem with Motoman Driver and a controller with more than 3 control groups (just a guess it can be another thing).

So my recommendations would be: 
1. Try ros-industrial/motoman#135, if that doesn't solve your issue, then...
2.  Go to `motoman/motoman_driver/src/simple_messages/joint_feedback_ex.cpp` and modify the following function:  `bool JointFeedbackEx::unload(industrial::byte_array::ByteArray *buffer)` and at the end of the "for" add
```
    if(i != 0){
        tmp_msg.init(j_feedback);
        this->joint_feedback_messages_.push_back(tmp_msg);
     }
```
(This is a solution posted on this [google group](https://groups.google.com/forum/m/#!topic/swri-ros-pkg-dev/9dPSw1mHDn8) )
3. If the previous things do not help, create an issue on the ros-industrial/motoman package, those guys are the true heroes.
     
PS: If you find a solution please don't forget to post it, e need to create a library of issues and information....good luck.

**Comment by Danfoa** [comment]
As far as I can see there is no errors in the .yaml conffiguration files and joint naming.

**Comment by SinaiAranda** [comment]
@Danfoa you can show me the roslaunch motoman_csda10f_support robot_interface_streaming_csda10f.launch controller:="dx100" robot_ip:="192.168.255.10" --screen

I have not been able to advance more than[ this](https://discourse.ros.org/t/using-the-motoman-sda20d-dual-arms-dx100-ros-interface-indigo/4667).

**Comment by SinaiAranda** [comment]
Thanks!
https://github.com/ros-industrial/motoman/pull/225#issuecomment-401814074

**Comment by gavanderhoorn** [comment]
Just to comment on this: several issues have been identified when trying to use the current `motoman_driver` nodes with robots such as yours.

One side has been fixed (ie: robot->driver), but the other side (driver->robot) has not. This means that receiving robot state should be ok now, but commanding motion needs some work.

I have a few fixes that I can make available to you. If you could test them with your setup, I would be grateful.

**Comment by SinaiAranda** [comment]
I've done some movement tests from Rviz and it seems to work. Of course, I can do it.

**Comment by gavanderhoorn** [comment]
Which specific fork/branch are you using?

**Comment by SinaiAranda** [comment]
These are [motoman driver](https://github.com/sinaiaranda-CIDESI/motoman_sda20d), [motoman_sda20d](https://github.com/sinaiaranda-CIDESI/motoman_sda20d) and Motoplus v180. I see that already there other new version 1.8.1.

**Comment by gavanderhoorn** [comment]
Your `motoman_driver` link points to your own repository (`sinaiaranda-CIDESI/motoman_sda20d`) and that doesn't contain the actual `motoman_driver` package.

I need info on `motoman_driver` alone. Exact repository and branch / version.

**Comment by SinaiAranda** [comment]
Sorry, is this [https://github.com/sinaiaranda-CIDESI/motoman_driver](https://github.com/sinaiaranda-CIDESI/motoman_driver)

**Comment by gavanderhoorn** [comment]
Is this a copy of the code in `ros-industrial/motoman`?

**Comment by SinaiAranda** [comment]
yes, I loaded it from my computer. The new has some modifications.

**Comment by SinaiAranda** [comment]
Just run with real robot, not on simulation.

**Comment by gavanderhoorn** [comment]
It's difficult to see exactly which changes you've made this way. Can you please fork `ros-industrial/motoman`, make a branch and upload your changes there?

**Comment by SinaiAranda** [comment]
I do not have changes of ros-industrial/motoman. First I used [this](https://github.com/Jmeyer1292/motoman/blob/9e7b50b860c47934011909d5854ea1d38dde6858/motoman_driver/src/simple_message/joint_feedback_ex.cpp#L110-L151) modification. After, you told me that motoman_driver kinetic already is correct. This [motoman_driver](https://github.com/ros-industrial/motoman) kinetic-devel I used for the real robot without problem just from Rviz. For this [roslaunch](https://github.com/sinaiaranda-CIDESI/motoman_sda20d/blob/master/motoman_sda20d_moveit_config/launch/robot_sim_sda20d.launch) (simulation) have this problem:

`[ WARN] [1530647782.085193845]: Waiting for sda20d/sda20d_r1_controller/joint_trajectory_action to come up
[ERROR] [1530647788.085494466]: Action client not connected: sda20d/sda20d_r1_controller/joint_trajectory_action
[ WARN] [1530647793.207681309]: Waiting for sda20d/sda20d_b1_controller/joint_trajectory_action to come up
[ WARN] [1530647799.207887742]: Waiting for sda20d/sda20d_b1_controller/joint_trajectory_action to come up
[ERROR] [1530647805.208096116]: Action client not connected: sda20d/sda20d_b1_controller/joint_trajectory_action
[ERROR] [1530647809.413972570]: Unable to connect to move_group action server 'move_group' within allotted time (30s)
[ WARN] [1530647810.328533462]: Waiting for sda20d/sda20d_r2_controller/joint_trajectory_action to come up
[ WARN] [1530647816.328974007]: Waiting for sda20d/sda20d_r2_controller/joint_trajectory_action to come up
[ERROR] [1530647822.329238068]: Action client not connected: sda20d/sda20d_r2_controller/joint_trajectory_action
`

**Comment by Danfoa** [comment]
I am currently on vacations and do not have access to the robot, so I am not able to provide any robot feedback any time soon, Sorry.

I would recommend checking the motoman repository solved issues page and to contact directly motoman support.

I am sorry I cannot be of much help.
